### PR TITLE
Made readme

### PR DESCRIPTION
**Pull request by eozturk2**
Made readme to document how to get started with the ROS environment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+### Working with Catkin
+Catkin is ROS's build system, which traverses every ROS package in a source directory and compiles them into 
+ROS modules. Every ROS project has a Catkin workspace structure:
+![Catkin workspace](https://miro.medium.com/v2/resize:fit:1400/format:webp/0*vfNM1mbkhUpvK-nW.png)
+
+This repository represents the "src" folder in this file tree. So, create a folder that will be your catkin
+workspace and clone this repository inside that folder. (This part might not be necessary, but rename the cloned
+folder "src" in case the catkin builder gives errors.)
+
+Navigate to the folder into which you cloned this repo, and run `catkin_make`. This should build the modules and
```
Comment (acoulombe) [inline README.md:30]: we don't use the `catkin_make` build structure. We use the subtly different ` catkin make`, which gives better development tools than the prior underscore variant that only builds when the make is invoked

File: README.md [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+create two more folders: "build" and "devel". At this point, you should source the setup script in the devel folder:
+
+`source devel/setup.bash`
+
+### Starting Nodes
+To start the nodes, you should first run `roscore` in a seperate terminal. This is the program that allows all
+ROS modules to communicate. Then, you can go back to the first terminal to start whichever node(s) you need.
+Navigate to the "rover" folder (or "src" if you renamed it), then use this syntax:
+
+`rosrun [package_name] [node_name]`
```
Comment (acoulombe) [inline README.md:40]: Also point to the launch files

File: README.md [hunk 3]
```diff
@@ -1,6 +1,43 @@
...
+
+### Working with Ubuntu
+To use ROS, you will need an Ubuntu system. For Windows users, options include dualbooting, virtual machines
+or WSL (Windows Subsystem for Linux). You can also try VPN'ing into McGill EMF (Engineering Microcomputing
+Facilities) PC's, but some of the steps will need sudo which might not be allowed.
+
+### Installing ROS
+Follow [these steps](http://wiki.ros.org/noetic/Installation/Ubuntu) to install ROS. Next, follow [these steps](http://wiki.ros.org/ROS/Tutorials/InstallingandConfiguringROSEnvironment) to see how to configure your shell and how to make a workspace. If you
+want, you can keep going with the tutorials to familiarize yourself with ROS and its commands. In addition, you
+should install pyserial and pybullet for Python.
```
Comment (acoulombe) [inline README.md:19]: we have a requirements.txt with all the python packages to install. Better to point to that than each individual package needed

**Review by acoulombe — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (eozturk2) [reply README.md:30]: Is it `catkin make` or `catkin build` ?

File: README.md (hunk 2 above)
Comment (eozturk2) [reply README.md:40]: Done

File: README.md (hunk 1 above)
Comment (eozturk2) [reply README.md:30]: `catkin make` doesn't seem to work on my end

Comment (eozturk2) [reply README.md:30]: Again, `catkin make` apparently isn't a thing. `catkin init` followed by a `catkin build` does work though. Another thing is that CameraData isn't a ROS module so it looks like we can't import stuff from it in the human control interface. This is what kills the GamepadProcess.py. For now I commented out the parts that need the camera orientation, and Shaswata said that CameraData might need to be another ROS module.

Comment (acoulombe) [reply README.md:30]: just checked again, it is the `catkin init` followed by `catkin build` development flow we are using. My mind is wandering. CameraData will either need to become a package.

Comment (eozturk2) [reply README.md:30]: Another problem with the launch files is that the embedded bridge doesn't seem to recognize some of the new arm module messages, or something like that. @Shaswata79 can probably clarify better than I can.

Comment (eozturk2) [reply README.md]: Fixed the readme too while I was at it

**Review by acoulombe — APPROVED** [review]

**Pull request merged**